src/utils/Polygons.ts

```diff
@@ -109,6 +111,34 @@ export class Polygon {
         return new Polygon(scaled);
     }
 
+    public static clip(a: Polygon, box: BoundingBox): Polygon {
+        if (a.empty) {
+            return new Polygon();
+        }
+
+        // Match all the numbers in the SVG using a regex, which is faster than scanning in Javascript
+        const re = /(\d+) (\d+)/g; // Adapted from here: https://stackoverflow.com/a/18085/871910
```

Comment:
I forget right now how Javascript handles compiling regex.  If we do it here inside the function, does it compile the regex every time we call the function, or does it cache the regex the first time it is compiled?  If it doesn't cache the regex, then maybe put it outside the function body so it gets compiled only once (same for the instance below).

Reply:
https://stackoverflow.com/a/8814176/871910 says regex literals are compiled once when the script is loaded. So we're good.

Reply:
Nice.